brain/detail/circuit.h

```diff
@@ -530,12 +530,22 @@ class SonataCircuit : public Circuit::Impl
         const size_t startIdx = *gids.begin();
         const size_t endIdx = *gids.rbegin() + 1;
 
-        const auto rotationAngleX =
-            nodeGroup.getAttribute<float>("rotation_angle_x", startIdx, endIdx);
-        const auto rotationAngleY =
-            nodeGroup.getAttribute<float>("rotation_angle_y", startIdx, endIdx);
-        const auto rotationAngleZ =
-            nodeGroup.getAttribute<float>("rotation_angle_z", startIdx, endIdx);
+        auto getter = [this](const char* attr, size_t start, size_t end) {
+            try
+            {
+                return nodeGroup.getAttribute<float>(attr, start, end);
+            }
+            catch (const HighFive::Exception& e)
```

Comment:
Is there any way to check if the attribute exists instead of throwing and catching exceptions? My issue here is that we might also catch exceptions we actually want to throw.

Reply:
Yes, but I won't do that. The alternative is to get a vector with all object names (datasets or not) and then do a linear search on that list. The resulting code would look less clear.